src/coreclr/jit/flowgraph.cpp

```diff
@@ -938,11 +940,34 @@ GenTreeCall* Compiler::fgGetStaticsCCtorHelper(CORINFO_CLASS_HANDLE cls, CorInfo
             result->gtCallMoreFlags |= GTF_CALL_M_HELPER_SPECIAL_DCE;
         }
     }
+    else if (bNeedClassID && !pmoduleID && !pclsID && (helper == CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE) &&
+        !compCurBB->isRunRarely() && (dataBlobOffset >= 0) && (static_cast<ssize_t>(clsID) >= 0))
+    {
+        assert(initFlag > 0);
 
+        // Inline "Is Class Initialized" check:
+        //
+        //   if (((UINT8*)moduleId + offsetof(DomainLocalModule, m_pDataBlob))[classId] & INITIALIZED_FLAG == 0)
+        //       return hellper(moduleId, classId);
```

Comment:
This helper is the rarely called path now. We should call `CORINFO_HELP_INITCLASS` that takes just a single `MethodTable*` argument to make the code smaller. `CORINFO_HELP_INITCLASS` is a bit slower than the helper that takes module and class IDs, but it does not matter since it will be called just once.